WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModel.swift

```diff
@@ -0,0 +1,297 @@
+                observePeriodViewModel()
+                Task { [weak self] in
+                    await self?.reloadData()
+                }
+            }
+            .store(in: &subscriptions)
+    }
+
+    func observePeriodViewModel() {
+        guard let periodViewModel else {
+            return
+        }
+
+        periodViewModel.timeRangeBarViewModel
+            .map { $0.timeRangeText }
+            .assign(to: &$timeRangeText)
+    }
+}
+
+// MARK: - Syncing data
+//
+private extension StorePerformanceViewModel {
+    @MainActor
+    func syncAllStats() async throws {
+        currentDate = Date()
+        let timezoneForSync = TimeZone.siteTimezone
+        let latestDateToInclude = timeRange.latestDate(currentDate: currentDate, siteTimezone: timezoneForSync)
+
+        try await withThrowingTaskGroup(of: Void.self) { group in
+            group.addTask { @MainActor [weak self] in
+                guard let self else { return }
+                statsVersion = await syncStats(latestDateToInclude: latestDateToInclude)
+            }
+
+            group.addTask { [weak self] in
+                try await self?.syncSiteVisitStats(latestDateToInclude: latestDateToInclude)
+            }
+
+            group.addTask { [weak self] in
+                try await self?.syncSiteSummaryStats(latestDateToInclude: latestDateToInclude)
+            }
+
+            while !group.isEmpty {
+                // rethrow any failure.
+                try await group.next()
+            }
+        }
+    }
+
+    /// Syncs store stats for dashboard UI.
+    @MainActor
+    func syncStats(latestDateToInclude: Date) async -> StatsVersion {
+        let waitingTracker = WaitingTimeTracker(trackScenario: .dashboardMainStats)
+        let earliestDateToInclude = timeRange.earliestDate(latestDate: latestDateToInclude, siteTimezone: siteTimezone)
+        return await withCheckedContinuation { continuation in
+            stores.dispatch(StatsActionV4.retrieveStats(siteID: siteID,
+                                                        timeRange: timeRange,
+                                                        timeZone: siteTimezone,
+                                                        earliestDateToInclude: earliestDateToInclude,
+                                                        latestDateToInclude: latestDateToInclude,
+                                                        quantity: timeRange.maxNumberOfIntervals,
+                                                        forceRefresh: true,
+                                                        onCompletion: { result in
+                switch result {
+                case .success:
+                    waitingTracker.end()
+                    continuation.resume(returning: StatsVersion.v4)
+                case .failure(let error):
+                    DDLogError("⛔️ Dashboard (Order Stats) — Error synchronizing order stats v4: \(error)")
```

Comment:
np: wonder if we should rewrite this to `Dashboard (Performance)`

Reply:
Updated the log in 383915910e8b39e93b7c64601e540a81cbc57708.